src/runtime_src/core/tools/common/ReportHost.cpp

```diff
@@ -109,13 +120,45 @@ ReportHost::writeReport(const xrt_core::device* /*_pDevice*/,
   _output << "Devices present\n";
   const boost::property_tree::ptree& available_devices = _pt.get_child("host.devices", empty_ptree);
 
-  if(available_devices.empty())
+  if (available_devices.empty())
     _output << "  0 devices found" << std::endl;
-  
-  for(auto& kd : available_devices) {
+
+  // Maps the data key to the maximum length amongst all devices
+  std::map<std::string, size_t> data_length_map;
+  for (auto& kd : available_devices) {
     const boost::property_tree::ptree& dev = kd.second;
+    updateDataMapValue(data_length_map, dev, "bdf");
+    updateDataMapValue(data_length_map, dev, "vbnv");
+    updateDataMapValue(data_length_map, dev, "id");
+    updateDataMapValue(data_length_map, dev, "instance");
+  }
+
+  // Allocate the header strings
+  std::string bdf_header = "BDF";
+  std::string vnbv_header = "Shell";
+  std::string id_header = "Interface UUID";
+  std::string instance_header = "Device ID";
+  // Verify the headers are not longer than the longest data string
+  data_length_map["bdf"] = std::max(data_length_map["bdf"], bdf_header.size());
+  data_length_map["vbnv"] = std::max(data_length_map["vbnv"], vnbv_header.size());
+  data_length_map["id"] = std::max(data_length_map["id"], id_header.size());
+  // Generate the number of blanks required for equal spacing
+  std::string bdf_blanks = std::string(data_length_map["bdf"] - bdf_header.size(), ' ');
+  std::string vnbv_blanks = std::string(data_length_map["vbnv"] - vnbv_header.size(), ' ');
+  std::string id_blanks = std::string(data_length_map["id"] - id_header.size(), ' ');
+
+  // Output the data headers in a table format
+  _output << boost::format("  %s%s : %s%s %s%s %s\n") % bdf_header % bdf_blanks % vnbv_header % vnbv_blanks % id_header % id_blanks % instance_header;
+
+  // Output each devices data in a table format
+  for (auto& kd : available_devices) {
+    const boost::property_tree::ptree& dev = kd.second;
+
     std::string note = dev.get<bool>("is_ready") ? "" : "NOTE: Device not ready for use";
-    _output << boost::format("  [%s] : %s %s %s\n") % dev.get<std::string>("bdf") % dev.get<std::string>("vbnv") % dev.get<std::string>("instance", "") % note;
+    bdf_blanks = std::string(data_length_map["bdf"] - dev.get<std::string>("bdf").size(), ' ');
+    vnbv_blanks = std::string(data_length_map["vbnv"] - dev.get<std::string>("vbnv").size(), ' ');
+    id_blanks = std::string(data_length_map["id"] - dev.get<std::string>("id").size(), ' ');
+    _output << boost::format("  %s%s : %s%s %s%s %s %s\n") % dev.get<std::string>("bdf") % bdf_blanks % dev.get<std::string>("vbnv") % vnbv_blanks % dev.get<std::string>("id") % id_blanks % dev.get<std::string>("instance", "") % note;

```

Comment:
I like what you are trying to do here and it is something that should be built upon.

In a nutshell, what you are doing is creating a table and making sure that the columns are right justified.  This functionality is literally screaming to be a helper class (e.g. _Table2D_) that will:

1. Allow users to add a 2D matrix of data
     e.g., `Table2D::addMatrix(std::vector<const std::vector<std::string>> & myMatrix)`
2. Instruct the justification on each column 
     e.g., `Table2D::justify(int column, enum Justification eJustification);`
3. Print out the table
     e.g., '`Table2D::writeTable(std::ostreamstring & oStream);`

This way, other parts of the code can take advantage of the work you have done.

Reply:
Sounds good!

Reply:
I added the Table2D class. Let me know what you think of the implementation